src/lib/mina_state/snarked_ledger_state.ml

```diff
@@ -476,7 +476,7 @@ module Make_str (A : Wire_types.Concrete) = struct
 
     let all bs = Tick.(Run.run_checked (Boolean.all bs))
 
-    let ( || ) b1 b2 = Tick.(Run.run_checked Boolean.(b1 || b2))
+    let ( || ) b1 b2 = Tick.(Run.run_checked Boolean.(b1 ||| b2))
```

Comment:
This change does not appear necessary nor related to this PR. Could you revert?

Unnecessary because the definition of `Boolean_intf` states 

```
(** Boolean or *)
  val ( || ) : var -> var -> var checked

  (** Boolean or, non-aliasing to [bool] operator. *)
  val ( ||| ) : var -> var -> var checked
```

Reply:
@rbonichon, actually it is necesary as without it `dune build --instrument-with bisect_ppx ...`  fails. Please see this PR for more context, which is also trying to solve this problem, but somehow it is not merged yet:

https://github.com/MinaProtocol/mina/pull/12893

Reply:
I see. I So here's my suggestion:
1. Check with @joaosreis and adopt a single solution since the 2 choices are obviously conflicting. Maybe build on top of his PR? FWIW. I would tend to prefer your solution.
2. However, add a comment here regarding the reason for the change. It looks like a bug in the bisect ppx. Could you check upstream? There was a new release 4 days ago too: could it make this issue go away?